### PR TITLE
[STG71]Blob API Diff

### PR DESCRIPTION
**Pull request by xiafu-msft**

### CODE REVIEW

File: sdk/storage/azure-storage-blob/out/azure/storage/blob/_blob_client.pyi [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    def stage_block_from_url(self, block_id: str, source_url: str, source_offset: Optional[int]=..., source_length: Optional[int]=..., source_content_md5: Optional[Union[bytes, bytearray]]=..., **kwargs: Any) -> None: ...
+    def get_block_list(self, block_list_type: Optional[str]=..., **kwargs: Any) -> Tuple[List[BlobBlock], List[BlobBlock]]: ...
+    def commit_block_list(self, block_list: List[BlobBlock], content_settings: Optional[ContentSettings]=..., metadata: Optional[Dict[str, str]]=..., **kwargs: Any) -> Dict[str, Union[str, datetime]]: ...
+    def set_premium_page_blob_tier(self, premium_page_blob_tier: Union[str, PremiumPageBlobTier], **kwargs: Any) -> None: ...
+    def get_page_ranges(self, offset: Optional[int]=..., length: Optional[int]=..., previous_snapshot_diff: Optional[Union[str, Dict[str, Any]]]=..., **kwargs: Any) -> Tuple[List[Dict[str, int]], List[Dict[str, int]]]: ...
+    def get_managed_disk_page_range_diff(self, offset: Optional[int]=..., length: Optional[int]=..., prev_snapshot_url: Optional[str]=..., **kwargs: Any) -> Tuple[List[Dict[str, int]], List[Dict[str, int]]]: ...
```
Comment (annatisch) [inline sdk/storage/azure-storage-blob/out/azure/storage/blob/_blob_client.pyi]: We would like to remove this API completely - as the feature can still be accessed via the existing `get_page_ranges` API.

File: sdk/storage/azure-storage-blob/out/azure/storage/blob/_blob_service_client.pyi [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    def from_connection_string(cls: Any, conn_str: str, credential: Optional[Any]=..., **kwargs: Any) -> BlobServiceClient: ...
+    def get_user_delegation_key(self, key_start_time: datetime, key_expiry_time: datetime, **kwargs: Any) -> UserDelegationKey: ...
+    def get_account_information(self, **kwargs: Any) -> Dict[str, str]: ...
+    def get_service_stats(self, **kwargs: Any) -> Dict[str, Any]: ...
+    def get_service_properties(self, **kwargs: Any) -> Dict[str, Any]: ...
+    def set_service_properties(self, analytics_logging: Optional[BlobAnalyticsLogging]=..., hour_metrics: Optional[Metrics]=..., minute_metrics: Optional[Metrics]=..., cors: Optional[List[CorsRule]]=..., target_version: Optional[str]=..., delete_retention_policy: Optional[RetentionPolicy]=..., static_website: Optional[StaticWebsite]=..., **kwargs: Any) -> None: ...
+    def list_containers(self, name_starts_with: Optional[str]=..., include_metadata: Optional[bool]=..., **kwargs: Any) -> ItemPaged[ContainerProperties]: ...
+    def create_container(self, name: str, metadata: Optional[Dict[str, str]]=..., public_access: Optional[Union[PublicAccess, str]]=..., **kwargs: Any) -> ContainerClient: ...
```
Comment (annatisch) [inline sdk/storage/azure-storage-blob/out/azure/storage/blob/_blob_service_client.pyi]: For the `create_container` API, we should rename the ContainerCpk parameter to `container_encryption_scope`

File: sdk/storage/azure-storage-blob/out/azure/storage/blob/_container_client.pyi [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+from typing import Any, AnyStr, Dict, IO, Iterable, Iterator, Optional, Union
+
+class ContainerClient(StorageAccountHostsMixin):
+    container_name: Any = ...
+    def __init__(self, account_url: str, container_name: str, credential: Optional[Any]=..., **kwargs: Any) -> None: ...
+    @classmethod
+    def from_container_url(cls: Any, container_url: str, credential: Optional[Any]=..., **kwargs: Any) -> ContainerClient: ...
+    @classmethod
+    def from_connection_string(cls: Any, conn_str: str, container_name: str, credential: Optional[Any]=..., **kwargs: Any) -> ContainerClient: ...
+    def create_container(self, metadata: Optional[Dict[str, str]]=..., public_access: Optional[Union[PublicAccess, str]]=..., **kwargs: Any) -> None: ...
```
Comment (annatisch) [inline sdk/storage/azure-storage-blob/out/azure/storage/blob/_container_client.pyi]: Same comment as above re create container

File: sdk/storage/azure-storage-blob/out/azure/storage/blob/_container_client.pyi [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    def get_account_information(self, **kwargs: Any) -> Dict[str, str]: ...
+    def get_container_properties(self, **kwargs: Any) -> ContainerProperties: ...
+    def set_container_metadata(self, metadata: Optional[Dict[str, str]]=..., **kwargs: Any) -> Dict[str, Union[str, datetime]]: ...
+    def get_container_access_policy(self, **kwargs: Any) -> Dict[str, Any]: ...
+    def set_container_access_policy(self, signed_identifiers: Dict[str, AccessPolicy], public_access: Optional[Union[str, PublicAccess]]=..., **kwargs: Any) -> Dict[str, Union[str, datetime]]: ...
+    def list_blobs(self, name_starts_with: Optional[str]=..., include: Optional[Any]=..., **kwargs: Any) -> ItemPaged[BlobProperties]: ...
+    def walk_blobs(self, name_starts_with: Optional[str]=..., include: Optional[Any]=..., delimiter: str=..., **kwargs: Optional[Any]) -> ItemPaged[BlobProperties]: ...
+    def upload_blob(self, name: Union[str, BlobProperties], data: Union[Iterable[AnyStr], IO[AnyStr]], blob_type: Union[str, BlobType]=..., length: Optional[int]=..., metadata: Optional[Dict[str, str]]=..., **kwargs: Any) -> BlobClient: ...
```
Comment (annatisch) [inline sdk/storage/azure-storage-blob/out/azure/storage/blob/_container_client.pyi]: For all the blob operations that take CpkScopeInfo, we should rename this parameter to `encryption_scope` and have it just take a string.

File: sdk/storage/azure-storage-blob/out/azure/storage/blob/_blob_client.pyi [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    container_name: Any = ...
+    blob_name: Any = ...
+    snapshot: Any = ...
+    def __init__(self, account_url: str, container_name: str, blob_name: str, snapshot: Optional[Union[str, Dict[str, Any]]]=..., credential: Optional[Any]=..., **kwargs: Any) -> None: ...
+    @classmethod
+    def from_blob_url(cls: Any, blob_url: str, credential: Optional[Any]=..., snapshot: Optional[Union[str, Dict[str, Any]]]=..., **kwargs: Any) -> BlobClient: ...
+    @classmethod
+    def from_connection_string(cls: Any, conn_str: str, container_name: str, blob_name: str, snapshot: Optional[str]=..., credential: Optional[Any]=..., **kwargs: Any) -> BlobClient: ...
+    def get_account_information(self, **kwargs: Any) -> Dict[str, str]: ...
+    def upload_blob(self, data: Union[Iterable[AnyStr], IO[AnyStr]], blob_type: Union[str, BlobType]=..., length: Optional[int]=..., metadata: Optional[Dict[str, str]]=..., **kwargs: Any) -> Any: ...
```
Comment (annatisch) [inline sdk/storage/azure-storage-blob/out/azure/storage/blob/_blob_client.pyi]: All the blob operations that take CpkInfo should take a string parameter `encryption_scope`

File: sdk/storage/azure-storage-blob/out/azure/storage/blob/_models.pyi [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+    def from_string(cls, permission: Any): ...
+
+class ContainerCpkScopeInfo(ContainerCpkScopeInfoGen):
+    default_encryption_scope: Any = ...
+    deny_encryption_scope_override: Any = ...
+    def __init__(self, **kwargs: Any) -> None: ...
+
+class CpkScopeInfo(CpkScopeInfoGen):
+    encryption_scope: Any = ...
+    def __init__(self, **kwargs: Any) -> None: ...
```
Comment (annatisch) [inline sdk/storage/azure-storage-blob/out/azure/storage/blob/_models.pyi]: We should remove this model completely and just pass around the `encryption_scope` string

File: sdk/storage/azure-storage-blob/out/azure/storage/blob/_models.pyi [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+    read: Any = ...
+    add: Any = ...
+    create: Any = ...
+    write: Any = ...
+    delete: Any = ...
+    def __init__(self, read: bool = ..., add: bool = ..., create: bool = ..., write: bool = ..., delete: bool = ...) -> None: ...
+    @classmethod
+    def from_string(cls, permission: Any): ...
+
+class ContainerCpkScopeInfo(ContainerCpkScopeInfoGen):
```
Comment (annatisch) [inline sdk/storage/azure-storage-blob/out/azure/storage/blob/_models.pyi]: This model should be renamed to `ContainerEncryptionScope`

File: sdk/storage/azure-storage-blob/out/azure/storage/blob/_models.pyi [hunk 8]
```diff
@@ -0,0 +1,231 @@
...
+    default_encryption_scope: Any = ...
+    deny_encryption_scope_override: Any = ...
```
Comment (annatisch) [inline sdk/storage/azure-storage-blob/out/azure/storage/blob/_models.pyi]: This property should be renamed to align with .NET/Java to `prevent_encryption_scope_override`

File: sdk/storage/azure-storage-blob/out/azure/storage/blob/_models.pyi (hunk 7 above)
Comment (annatisch) [reply sdk/storage/azure-storage-blob/out/azure/storage/blob/_models.pyi]: Additionally - this object should be returned for `get_container_properties`

File: sdk/storage/azure-storage-blob/out/azure/storage/blob/_blob_client.pyi (hunk 1 above)
Comment (xiafu-msft) [reply sdk/storage/azure-storage-blob/out/azure/storage/blob/_blob_client.pyi]: We haven't reached an agreement to merge it across all SDKs, probably leave the managed_disk_diff API here for now

**Pull request closed without merging**